stumpy/core.py

```diff
@@ -1524,19 +1581,31 @@ def mass(
     if np.any(~np.isfinite(Q)):
         distance_profile[:] = np.inf
     else:
-        if M_T is None or Σ_T is None:
-            T, M_T, Σ_T = preprocess(T, m)
+        if M_T is None or Σ_T is None or T_subseq_isconstant is None:
```

Comment:
> Note that I would put `rolling_isconstant` AFTER `np.inf` are set to `np.nan`. This way, we know that we are dealing with `np.nan` only when checking `isconstant`

It seems `core.rolling_isconstant(...)` can handle both `inf` and `nan`.  It seems `compute_mean_std` can also handle these two non-finite values. So, why not do `T[~np.isfinite(T)] = 0` at the end? 

This may help others develop the code more easily in the future as they know they should calculate `M_T`, `Σ_T`, and `T_subseq_isconstant` in the very beginning of the code before cleaning `T`.

Thoughts?

Reply:
>  It seems compute_mean_std can also handle these two non-finite values.

No, there is a subtle difference here. When you do `np.nanmean([1.0, 2.0, np.nan])` you get `1.5`. However, if you do `np.nanmean([1.0, 2.0, np.inf])` you get `np.inf`!  The behavior is very different and (I think) the ordering matters.

Reply:
Correct. Just one last thing:

I think `core.compute_mean_std` takes care of this matter. For example:

```
a = np.array([np.inf, 1, 2, 3, np.nan])
core.compute_mean_std(a, 3)[0]

>>> array([inf,  2., inf])
```

But, I feel what you did is better as it increases the safety. I am going to close this. If you change your mind, please let me know.

Reply:
> btw, since compute_mean_std can take care of things (as shown in the example above), we may simplify the cleaning process in function core.py: preprocess(...)

I disagree. `compute_mean_std` should really only be responsible for computing the mean and stddev. It feels "weird" to assume that it can "take care of things". It feels like it is unnecessarily increasing the scope of the function. 

Or to put it differently, with the cost being so low, I would rather than be explicit rather than implicit. So, I make sure to explicitly convert all my `np.inf` into `np.nan`. Basically, the assumption is that `stumpy` hates `np.inf` and, if your data contains `np.inf` then some things might not work. Also, IIRC, np.inf may be handled differently on different hardware architectures so while it might work on your machine, it may not work consistently elsewhere.

Reply:
Thanks for the elaboration! I get it now!